### ISSUE TITLE
Split "ClassAndModuleChildren" rule in 2 rules

### ISSUE THREAD

**Issue by WaKeMaTTa** [rubocop/rubocop#12851]
## Is your feature request related to a problem? Please describe.

In my company we use this rules:

- Use always plural for namespaces. This namespaces need to be `Module`s.
- Use always singular for object. This object need to be `Class`es.

The result is:

```ruby
module Companies::Products
  class TagService
  end
end
```

Then then rubocop is always complaying. The rubocop rule that complains is `ClassAndModuleChildren`

## Describe the solution you'd like

I would like to split the rubocop rule `Style/ClassAndModuleChildren` in 2.

Having `Style/ClassChildren` and `Style/ModuleChildren`, it would allow me to define it this way:

```yml
Style/ModuleChildren:
  EnforcedStyle: "compact"
Style/ClassChildren:
  EnforcedStyle: "nested"
```
